src/plugins/data/public/search/types.ts

```diff
@@ -84,3 +98,11 @@ export interface SearchServiceStartDependencies {
   fieldFormats: AggsStartDependencies['fieldFormats'];
   indexPatterns: DataViewsContract;
 }
+
+export interface SearchResponseWarning {
+  type: 'timed_out' | 'generic_shard_warning' | estypes.ShardFailure['reason']['reason'];
```

Comment:
we may want to refine the type definition after https://github.com/elastic/elasticsearch/pull/89252